### PR TITLE
Feat(129): Sentry expanded context

### LINKED ISSUE #129: [Improvement] Improve error handling
**Description**

We've seen that when the application throws an error it is not so easy to determine the cause. We should update all error handling code to include as much information as possible without causing the application to crash. 

The information we should try and add should include the following:
- What may have caused the error (e.g. calling airgrab.claim() ...)
- Where the error occurred
- Context. Additional information that can help diagnose the issue (e.g. address of connected account, connected chain, provider info etc)
- Sentry integration. Push error logs and info to sentry

### PR DESCRIPTION
**Pull request by RyRy79261**
## Description

This PR expands Sentry's available context to include certain implementation specific properties

## Review guide

A hook for simply managing the context which is bundled into Sentry's injected logger was added, this PR did not focus on getting uniform coverage of Sentry reporting.

This is due to Sentry's next js already, auto-magically, reporting with indepth insights, any instance of `throw Error` 

It is recommended that a review of existing reporting is done, to cover gaps rather than blanket coat the code base in redundant reporting. 

## Related tickets

- closes #129

### CODE REVIEW

File: src/lib/fractal.ts [hunk 1]
```diff
@@ -112,6 +112,7 @@ export async function refetchKycStatus(
     return kycStatus;
   } catch (error) {
     // TODO: sentrify
+
```
Comment (chapati23) [inline src/lib/fractal.ts:115]: noticed the TODO comment above this line, did you mean to actually sentrify this part or just a formatting empty line?

**Review by chapati23 — COMMENTED** [review]
just two questions

**Comment by denviljclarke** [comment]
I'm taking over this ticket, but I will open a separate PR when ready. Aiming for tomorrow

File: src/lib/fractal.ts (hunk 1 above)
Comment (RyRy79261) [reply src/lib/fractal.ts:115]: @chapati23 So this is where @denviljclarke will be continuing the intergrations

File: src/hooks/use-sentry-context.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import { useCallback, useEffect, useState } from "react";
+import { Address } from "viem";
+import * as Sentry from "@sentry/nextjs";
+
+const CHAIN_CONTEXT = "web3-data";
```
[the remark this answers is not in the document]
Comment (RyRy79261) [reply src/hooks/use-sentry-context.tsx:7]: @chapati23 This is for Sentry to index certain contexts, I'm only injecting Web3 related ones with this, and clearing them as well.

If we wanted more say Browser related, locale related etc, then we would use that key for those domains.

**Pull request closed without merging**